climada/util/coordinates.py

```diff
@@ -76,6 +78,256 @@
+        belonging to its corresponding line (incl. line length of original geom.
+        and multi-index referring to original indexing)
+        
+    See also
+    --------
+    * util.coordinates.dist_great_circle_allgeoms()
+    * util.lines_polys_handler.point_exposure_from_lines()
+    """
+
+    if not isinstance(gdf_lines, gpd.GeoDataFrame):
+        gdf_lines = gpd.read_file(gdf_lines)
+        
+    gdf_points = gdf_lines.copy()
+    gdf_points['length'] = np.ones(len(gdf_points))*point_dist
+    gdf_points['length_full'] = dist_great_circle_allgeoms(gdf_points)
+
+    # split line lengths into relative fractions acc to point_dist (e.g. 0, 0.5, 1)
+    gdf_points['distance_vector'] = gdf_points.apply(
+        lambda row: np.linspace(0, 1, num=int(np.ceil(row.length_full/
+                                                      row.length)+1)), axis=1)
+
+    # create MultiPoints along the line for every position in distance_vector
+    gdf_points['geometry'] = gdf_points.apply(
+        lambda row: MultiPoint(
+            [row.geometry.interpolate(dist, normalized=True) 
+             for dist in row.distance_vector]), axis=1)
+    
+    # expand gdf from MultiPoint entries to single Points per row
+    return gdf_points.explode().drop(['distance_vector', 'length_full'], axis=1)
+
+def interpolate_polygons(gdf_poly, area_point):
+    """For a GeoDataFrame with polygons, get equally distributed lat/lon pairs
+    throughout the geometries, at a user-specified area distance
+    
+    Parameters
+    ----------
+    gdf_poly : (gpd.GeoDataFrame) with polygons to be interpolated
+    area_point : area in m2 which one point should represent
+    
+    Returns
+    -------
+    (gpd.GeoDataFrame) with individual Point per row, retaining all other column infos
+        belonging to its corresponding polygon
+    
+    See also
+    --------
+    * util.coordinates.dist_great_circle_allgeoms()
+    * util.lines_polys_handler.point_exposure_from_polygons()
+    """
+    
+    metre_dist = math.sqrt(area_point)
+    
+    gdf_poly['degree_dist'] = gdf_poly.apply(lambda row: metres_to_degrees(
+        row.geometry.representative_point().x, 
+        row.geometry.representative_point().y,
+        metre_dist), axis=1)
+    
+    # get params to make an even grid with desired resolution 
+    # over bounding boxes of polygons:
+    trans = gdf_poly.apply(lambda row: pts_to_raster_meta(
+        row.geometry.bounds, (row.degree_dist, -row.degree_dist)), axis=1)
+    gdf_poly['trans'] = pd.DataFrame(trans.tolist(), index=trans.index).iloc[:,2]
+   
+    gdf_poly['width'] = np.floor(abs(gdf_poly.geometry.bounds.minx-
+                                     gdf_poly.geometry.bounds.maxx)/
+                                 gdf_poly['degree_dist'])
+    gdf_poly['height'] = np.floor(abs(gdf_poly.geometry.bounds.miny-
+                                      gdf_poly.geometry.bounds.maxy)/
+                                  gdf_poly['degree_dist'])
```

Comment:
Why do you manually recompute width and height even though this is already returned by `pts_to_raster_meta`?

Reply:
Don't know why I did this, either ;). Probably temporal code-blindness.
Thanks.